### PR TITLE
Search user by username with send message test.

### PR DESCRIPTION
**Pull request by gmardok**
Search user by username with send message test.

### CODE REVIEW

File: TLSharp.Tests/TLSharpTests.cs [hunk 1]
```diff
@@ -346,5 +347,42 @@ private void GatherTestConfiguration()
...
+
+                user = contacts.users.lists
+                    .Where(x => x.GetType() == typeof(TLUser))
+                    .Cast<TLUser>()
+                    .FirstOrDefault(x => x.username == UserNameToSendMessage.TrimStart('@'));
+            }
+
+            if (user == null)
+            {
+                throw new System.Exception("Usename was not found: " + UserNameToSendMessage);
```
Comment (knocte) [inline TLSharp.Tests/TLSharpTests.cs]: Typo: Usename -> Username

File: TLSharp.Tests/TLSharpTests.cs [hunk 2]
```diff
@@ -346,5 +347,42 @@ private void GatherTestConfiguration()
...
+                .Cast<TLUser>()
+                .FirstOrDefault(x => x.username == UserNameToSendMessage.TrimStart('@'));
+
+            if (user == null)
+            {
+                var contacts = await client.GetContactsAsync();
+
+                user = contacts.users.lists
+                    .Where(x => x.GetType() == typeof(TLUser))
+                    .Cast<TLUser>()
```
Comment (knocte) [inline TLSharp.Tests/TLSharpTests.cs]: Please use `OfType<T>` instead of `Where().Cast()`

File: TLSharp.Tests/TLSharpTests.cs [hunk 3]
```diff
@@ -346,5 +347,42 @@ private void GatherTestConfiguration()
...
+
+            var client = NewClient();
+
+            await client.ConnectAsync();
+
+            var result = await client.SearchUserAsync(UserNameToSendMessage);
+
+            var user = result.users.lists
+                .Where(x => x.GetType() == typeof(TLUser))
+                .Cast<TLUser>()
```
Comment (knocte) [inline TLSharp.Tests/TLSharpTests.cs]: Please use `OfType<T>` instead of `Where().Cast()`.

**Comment by gmardok** [comment]
@knocte if you look through test you will see that everywhere Cast used.
Changes done.

**Comment by knocte** [comment]
> @knocte if you look through tests you will see that everywhere Cast used.

Not in latest master :) You forked the repo a bit before those changes ;)

**Pull request merged**